exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java

```diff
@@ -131,6 +144,54 @@ public QueryResult run(final WorkManager workManager, final WebUserConnection we
     return new QueryResult(queryId, webUserConnection, webUserConnection.results);
   }
 
+  private void applyOptions(WebUserConnection webUserConnection) throws BadRequestException {
+    if (options != null && !options.isEmpty()) {
+      SessionOptionManager sessionOptionManager = webUserConnection.getSession().getOptions();
+      for (Map.Entry<String, Object> entry : options.entrySet()) {
```

Comment:
Beyond the scope of this PR, but a better approach would be to implement a `setOption(String name, Object value)` method in `BaseOptionManager`. Then, build on that with a `setOptions<Map<String, Object>)` method. Then, we reuse these methods in the several other places where we do such things. Filed DRILL-7611 to track this request.

Reply:
Hmm, maybe.  I'm not sure that would ever be used anywhere else but in here, though.  Could probably hold off on that refactoring until you have at least two places where it would be useful.